frontend/src/components/NumberInput/NumberInput.tsx

```diff
@@ -88,31 +94,65 @@ export const NumberInput = forwardRef<NumberInputProps, 'input'>(
       : undefined
 
+    const inputBox = (hasPrefix: boolean) => (
+      <chakra.input
+        {...inputProps}
+        paddingInlineEnd={inputEndPadding}
+        // This flag should be set for form input fields, to prevent refresh on enter if form only has one input
+        {...(preventDefaultOnEnter
+          ? {
+              onKeyDown: (e) => {
+                if (e.key === 'Enter') {
+                  e.preventDefault()
+                }
+              },
+            }
+          : {})}
+        // Passing in ref to the input element so that it can be focused by
+        // the parent.
+        // No point passing the ref to the div wrapper as the main component
+        // is this input.
+        ref={inputRef}
+        __css={styles.field}
+        borderLeftRadius={hasPrefix ? 0 : undefined}
+      />
+    )
+
+    // TODO: replace with dynamic country loading if/when more currencies are added
+    const country = 'SG'
+
     return (
       <Box {...htmlProps} __css={styles.root}>
         {/* Using base input wrapper instead of `Input` component as the Input 
         component strips out some props such as `aria-invalid`, resulting in
         incorrect styling */}
-        <chakra.input
-          {...inputProps}
-          paddingInlineEnd={inputEndPadding}
-          // This flag should be set for form input fields, to prevent refresh on enter if form only has one input
-          {...(preventDefaultOnEnter
-            ? {
-                onKeyDown: (e) => {
-                  if (e.key === 'Enter') {
-                    e.preventDefault()
-                  }
-                },
-              }
-            : {})}
-          // Passing in ref to the input element so that it can be focused by
-          // the parent.
-          // No point passing the ref to the div wrapper as the main component
-          // is this input.
-          ref={inputRef}
-          __css={styles.field}
-        />
+        {props.prefix ? (
+          <InputGroup>
+            <InputLeftAddon
+              aria-disabled={inputProps.disabled}
+              as="label"
+              sx={styles.country}
+              background="transparent"
+              borderColor="neutral.400"
+            >
+              <HStack align="center" spacing={2}>
+                <Icon
+                  // Show Flags if available. If value does not exist for any reason,
+                  // a default fallback icon will be used by ChakraUI.
+                  // See https://chakra-ui.com/docs/media-and-icons/icon#fallback-icon.
+                  as={Flags[country]}
+                  role="img"
+                  aria-label={`Only ${country} numbers are allowed`}
+                  __css={styles.icon}
+                />
+                <Text>SGD</Text>
+              </HStack>
+            </InputLeftAddon>
+            {inputBox(true)}
+          </InputGroup>
+        ) : (
+          inputBox(false)
+        )}
```

Comment:
If we want a money input, would it be more appropriate to create a separate component for `MoneyInput` instead of trying to bake the functionality into `NumberInput`? Adding a specific `InputLeftAddon` feels somewhat outside the scope of what `NumberInput` is supposed to be for.

Reply:
Makes sense! Will make a separate component for money 👍

Reply:
Note: I've separated the component used into MoneyInput, but the currency flag and prefix is currently hard-coded in the component itself to be the Singapore flag and "SGD" respectively. ([b9cff23](https://github.com/opengovsg/FormSG/pull/5852/commits/b9cff23c97ebc6ae9f4b8f7dd306075e0f2e3e2c))